### PR TITLE
Fix segfault rz_bin_elf_init_dynamic_section

### PR DESCRIPTION
**Pull request by ghost**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Fix the segfault fuzz test.

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

**Comment by XVilka** [comment]
One test left:
```
[XX] /home/runner/work/rizin/rizin/test/bins/fuzzed <fuzz> /home/runner/work/rizin/rizin/test/bins/fuzzed/elf1
RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc aaa /home/runner/work/rizin/rizin/test/bins/fuzzed/elf1
-- stdout

-- stderr

-- exit status: -1
[**]     /home/runner/work/rizin/rizin/test/bins/fuzzed      504 OK         0 BR        1 XX        0 FX
```

**Comment by XVilka** [comment]
Now another test broken on FreeBSD:
```
[XX] db/formats/elf/versioninfo ELF: versioninfo 1
RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc iV bins/elf/analysis/serial
-- stdout
--- expected
+++ actual
@@ -1,15 +1,7 @@
 Version symbols has 11 entries:
  Addr: 0x00400442  Offset: 0x00000442
   0x00000000: 0 (*local*)
-  0x00000001: 2 (GLIBC_2.2.5)
-  0x00000002: 2 (GLIBC_2.2.5)
-  0x00000003: 2 (GLIBC_2.2.5)
-  0x00000004: 2 (GLIBC_2.2.5)
-  0x00000005: 2 (GLIBC_2.2.5)
   0x00000006: 3 (GLIBC_2.4)
-  0x00000007: 2 (GLIBC_2.2.5)
-  0x00000008: 2 (GLIBC_2.2.5)
-  0x00000009: 2 (GLIBC_2.2.5)
   0x0000000a: 0 (*local*)
 
```

**Comment by ghost** [comment]
> Now another test broken on FreeBSD:
> 
> ```
> [XX] db/formats/elf/versioninfo ELF: versioninfo 1
> RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc iV bins/elf/analysis/serial
> -- stdout
> --- expected
> +++ actual
> @@ -1,15 +1,7 @@
>  Version symbols has 11 entries:
>   Addr: 0x00400442  Offset: 0x00000442
>    0x00000000: 0 (*local*)
> -  0x00000001: 2 (GLIBC_2.2.5)
> -  0x00000002: 2 (GLIBC_2.2.5)
> -  0x00000003: 2 (GLIBC_2.2.5)
> -  0x00000004: 2 (GLIBC_2.2.5)
> -  0x00000005: 2 (GLIBC_2.2.5)
>    0x00000006: 3 (GLIBC_2.4)
> -  0x00000007: 2 (GLIBC_2.2.5)
> -  0x00000008: 2 (GLIBC_2.2.5)
> -  0x00000009: 2 (GLIBC_2.2.5)
>    0x0000000a: 0 (*local*)
>  
> ```

Yes, I am working on this.

File: librz/bin/format/elf/elf_symbols.c [hunk 1]
```diff
@@ -172,6 +172,10 @@ static ut64 get_index_from_buckets(ELFOBJ *bin, ut64 *bucket_offset, ut64 number
 
 	for (size_t i = 0; i < number_of_bucket; i++) {
 		ut64 tmp = BREAD32(bin->b, *bucket_offset);
```
Comment (ret2libc) [inline librz/bin/format/elf/elf_symbols.c]: why ut64 instead of ut32?

Comment (ghost) [reply librz/bin/format/elf/elf_symbols.c]: Good question, I will use ut32

**Review by XVilka — APPROVED** [review]

**Pull request merged**